ros2/test/pluginlib/ament_setup_library_tests.cc

```diff
@@ -0,0 +1,95 @@
+// Copyright 2023 Wouter van der Stoel
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "console_bridge/console.h"
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+#include "pluginlib/class_loader.hpp"
+#include "rcutils/logging.h"
+
+#include "ros2/test/pluginlib/regular_polygon.h"
+#include "ros2/test/pluginlib/square_ament_setup/ament_setup.h"
+#include "ros2/test/pluginlib/triangle_ament_setup/ament_setup.h"
+
+using ::testing::DoubleNear;
+using ::testing::Eq;
+
+class TestAmentSetup : public ::testing::Test {
+ public:
+  void SetUp() override {
```

Comment:
You can just use http://google.github.io/googletest/reference/testing.html#Test::SetUpTestSuite.